### PR TITLE
Added RBAC access acls/roles to service

### PR DESCRIPTION
**Pull request by hsong-rh**
This is the second phase to enable RBAC support in Approval service:

1. create `RBAC::Access` to handle access control list (acls) based on predefined permissions and resource definitions;
2. set role (`admin, approver, owner`) according to acls;

### CODE REVIEW

File: lib/rbac/permissions.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module RBAC
+  module Permissions
```
Comment (bzwei) [inline lib/rbac/permissions.rb]: why there are only four permissions?

File: lib/rbac/access.rb [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        @acls.any? do |item|
+          item.resource_definitions.any? do |rd|
+            @admin = true if rd.attribute_filter.key == 'id' &&
+                             rd.attribute_filter.operation == 'equal' &&
+                             rd.attribute_filter.value == '*'
+          end
+        end
+
+        approver_regexp = Regexp.new(":(workflows|\\*):(approve|\\*)")
+        @approver_acls = RBAC::Service.paginate(api, :get_principal_access, {}, @app_name).select do |item|
```
Comment (bzwei) [inline lib/rbac/access.rb]: should not call `RBAC::Service` twice. Reuse the result from line 25

File: lib/rbac/permissions.rb (hunk 1 above)
Comment (hsong-rh) [reply lib/rbac/permissions.rb]: Other permissions are predefined.

File: lib/rbac/acls.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      end
+      acls
+    end
+
+    def add(acls, resource_id, permissions)
+      new_acls = []
+      permissions.each do |permission|
+        next if find_matching(acls, resource_id, permission)
+
+        new_acls << create_acl(permission, resource_id)
```
[the remark this answers is not in the document]
Comment (hsong-rh) [reply lib/rbac/acls.rb]: Need to remove duplicates.

File: lib/rbac/access.rb [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+    def process
+      RBAC::Service.call(RBACApiClient::AccessApi) do |api|
+        Rails.logger.info("Fetch access list for #{@app_name}")
+        full_acls = RBAC::Service.paginate(api, :get_principal_access, {}, @app_name)
+
+        approver_regexp = Regexp.new(":(workflows|\\*):(approve|\\*)")
+
+        full_acls.each do |item|
+          Rails.logger.info("Found ACL: #{item}")
```
Comment (bzwei) [inline lib/rbac/access.rb]: do you want to log every ACL? Perhaps debug is better.

File: lib/rbac/access.rb [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+      @approver_acls = []
+      @owner_acls    = []
+    end
+
+    def process
+      RBAC::Service.call(RBACApiClient::AccessApi) do |api|
+        Rails.logger.info("Fetch access list for #{@app_name}")
+        full_acls = RBAC::Service.paginate(api, :get_principal_access, {}, @app_name)
+
+        approver_regexp = Regexp.new(":(workflows|\\*):(approve|\\*)")
```
Comment (bzwei) [inline lib/rbac/access.rb]: Create two regexes in one place. Either in constructor or here.

**Pull request merged**